src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py

```diff
@@ -4736,3 +4736,124 @@ def test_sql_mi_failover_mgmt(self):
+                 '-u {admin_login} -p {admin_password} --subnet {subnet_id} --license-type {license_type} --capacity {v_cores} --storage {storage_size_in_gb} --edition {edition} --family {family}',
+                 checks=[
+                     self.check('name', '{managed_instance_name}'),
+                     self.check('resourceGroup', '{resource_group}'),
+                     self.check('administratorLogin', '{admin_login}'),
+                     self.check('vCores', '{v_cores}'),
+                     self.check('storageSizeInGb', '{storage_size_in_gb}'),
+                     self.check('licenseType', '{license_type}'),
+                     self.check('sku.tier', '{edition}'),
+                     self.check('sku.family', '{family}'),
+                     self.check('sku.capacity', '{v_cores}'),
+                     JMESPathCheck('identity', None)]).get_output_in_json()
+
+        managed_database_name = 'logReplayTestDb'
+        managed_database_name1 = 'logReplayTestDb1'
+        self.kwargs.update({
+            'managed_database_name': managed_database_name,
+            'managed_database_name1': managed_database_name1,
+            'storage_sas': 'sv=2019-02-02&ss=b&srt=sco&sp=rl&se=2023-12-02T00:09:14Z&st=2019-11-25T16:09:14Z&spr=https&sig=92kAe4QYmXaht%2FgjocUpioABFvm5N0BwhKFrukGw41s%3D',
+            'storage_uri': 'https://mijetest.blob.core.windows.net/pcc-remote-replicas-test',
+            'last_backup_name': 'log1.bak'
+        })
+
+        # Start Log Replay Service
+        self.cmd('sql midb log-replay start -g {resource_group} --mi {managed_instance_name} -n {managed_database_name} --ss {storage_sas} --su {storage_uri} --no-wait',
+                 checks=NoneCheck())
+
+        if self.in_recording or self.is_live:
+            sleep(10)
+
+        # Complete log replay service
+        self.cmd('sql midb log-replay complete -g {resource_group} --mi {managed_instance_name} -n {managed_database_name} --last-bn {last_backup_name}',
+                 checks=NoneCheck())
+
+        if self.in_recording or self.is_live:
+            sleep(60)
+
+        # Verify status is Online
+        self.cmd('sql midb show -g {resource_group} --mi {managed_instance_name} -n {managed_database_name}',
+                 checks=[
+                     JMESPathCheck('status', 'Online')])
+
+        # Cancel test for Log replay
+
+        # Start Log Replay Service
+        self.cmd('sql midb log-replay start -g {resource_group} --mi {managed_instance_name} -n {managed_database_name1} --ss {storage_sas} --su {storage_uri} --no-wait',
+                 checks=NoneCheck())
+
+        # Wait a minute to start restoring
+        if self.in_recording or self.is_live:
+            sleep(60)
+
+        # Cancel log replay service
+        self.cmd('sql midb log-replay stop -g {resource_group} --mi {managed_instance_name} -n {managed_database_name1} --yes',
```

Comment:
could you also add test for wait command?

Reply:
I didn't add wait command in test because the provisioning of MI takes too much time.. I did test it manual and it works as expected.
Hope this is ok